### PR TITLE
ref(rules): Update typing in rule files

### PR DESCRIPTION
**Pull request by ceorourke**
Pulls out the non-functional changes from https://github.com/getsentry/sentry/pull/68517 which are mostly typing related.

### CODE REVIEW

File: src/sentry/rules/processing/processor.py [hunk 1]
```diff
@@ -236,8 +245,9 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
                 if not predicate_func(predicate_iter):
                     return
             else:
-                self.logger.error(
-                    f"Unsupported {name}_match {match!r} for rule {rule.id}",
+                log_string = f"Unsupported {name}_match {match!r} for rule {rule.id}"
```
Comment (nhsiehgit) [inline src/sentry/rules/processing/processor.py:248]: does this need to be in a separate var?

**Review by nhsiehgit — APPROVED** [review]
👍

File: src/sentry/rules/processing/processor.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/rules/processing/processor.py:248]: I got some error if i didn't: `LOG011 avoid pre-formatting log messages`

**Review by saponifi3d — APPROVED** [review]

File: src/sentry/rules/conditions/event_frequency.py [hunk 2]
```diff
@@ -319,7 +346,7 @@ class EventFrequencyCondition(BaseEventFrequencyCondition):
     label = "The issue is seen more than {value} times in {interval}"
 
     def query_hook(
-        self, event: GroupEvent, start: datetime, end: datetime, environment_id: str
+        self, event: GroupEvent, start: datetime, end: datetime, environment_id: int
```
[the remark this answers is not in the document]
Comment (ceorourke) [reply src/sentry/rules/conditions/event_frequency.py:349]: I think this was just wrong before

**Review by JoshFerge — COMMENTED** [review]
i notice there is some test coverage lacking on these changes -- worth adding some now?

**Pull request merged**